diff --git a/p4-switch/sai-p4-target/Makefile.am b/p4-switch/sai-p4-target/Makefile.am
index 153b06c..53d7e07 100644
--- a/p4-switch/sai-p4-target/Makefile.am
+++ b/p4-switch/sai-p4-target/Makefile.am
@@ -42,7 +42,8 @@ AM_CXXFLAGS += -std=c++11 \
 -I$(top_srcdir)/@P4_BM_DIR@/include \
 -isystem$(top_srcdir)/@P4_BM_DIR@/third_party/jsoncpp/include \
 -isystem$(top_srcdir)/@P4_BM_DIR@/third_party/spdlog \
--DBMLOG_DEBUG_ON -DBMLOG_TRACE_ON
+-DBMLOG_DEBUG_ON -DBMLOG_TRACE_ON \
+-DFORCE_BOOST_SMART_PTR
 
 libsimpleswitch_la_LIBADD = \
 $(top_srcdir)/@P4_BM_DIR@/src/bm_sim/libbmsim.la \

